pkg/pn/info_test.go

```diff
@@ -0,0 +1,365 @@
+		{
+			Raw: "aFamily^aGiven^aMiddle^aPrefix^aSuffix=" +
+				"iFamily^iGiven^iMiddle^iPrefix^iSuffix" +
+				"",
+			Expected: Info{
+				Alphabetic: GroupInfo{
+					Raw:        "aFamily^aGiven^aMiddle^aPrefix^aSuffix",
+					FamilyName: "aFamily",
+					GivenName:  "aGiven",
+					MiddleName: "aMiddle",
+					NamePrefix: "aPrefix",
+					NameSuffix: "aSuffix",
+				},
+				Ideographic: GroupInfo{
+					Raw:        "iFamily^iGiven^iMiddle^iPrefix^iSuffix",
+					FamilyName: "iFamily",
+					GivenName:  "iGiven",
+					MiddleName: "iMiddle",
+					NamePrefix: "iPrefix",
+					NameSuffix: "iSuffix",
+				},
+				Phonetic: GroupInfo{
+					Raw:        "",
+					FamilyName: "",
+					GivenName:  "",
+					MiddleName: "",
+					NamePrefix: "",
+					NameSuffix: "",
+				},
+			},
+			RemoveTrailingEmpty: true,
+			IsEmpty:             false,
+		},
+		// No Ideographic
+		{
+			Raw: "aFamily^aGiven^aMiddle^aPrefix^aSuffix=" +
+				"=" +
+				"",
+			Expected: Info{
+				Alphabetic: GroupInfo{
+					Raw:        "aFamily^aGiven^aMiddle^aPrefix^aSuffix",
+					FamilyName: "aFamily",
+					GivenName:  "aGiven",
+					MiddleName: "aMiddle",
+					NamePrefix: "aPrefix",
+					NameSuffix: "aSuffix",
+				},
+				Ideographic: GroupInfo{
+					Raw:        "",
+					FamilyName: "",
+					GivenName:  "",
+					MiddleName: "",
+					NamePrefix: "",
+					NameSuffix: "",
+				},
+				Phonetic: GroupInfo{
+					Raw:        "",
+					FamilyName: "",
+					GivenName:  "",
+					MiddleName: "",
+					NamePrefix: "",
+					NameSuffix: "",
+				},
+			},
+			RemoveTrailingEmpty: false,
+			IsEmpty:             false,
+		},
+		// No Ideographic, no seps
+		{
+			Raw: "aFamily^aGiven^aMiddle^aPrefix^aSuffix",
+			Expected: Info{
+				Alphabetic: GroupInfo{
+					Raw:        "aFamily^aGiven^aMiddle^aPrefix^aSuffix",
+					FamilyName: "aFamily",
+					GivenName:  "aGiven",
+					MiddleName: "aMiddle",
+					NamePrefix: "aPrefix",
+					NameSuffix: "aSuffix",
+				},
+				Ideographic: GroupInfo{
+					Raw:        "",
+					FamilyName: "",
+					GivenName:  "",
+					MiddleName: "",
+					NamePrefix: "",
+					NameSuffix: "",
+				},
+				Phonetic: GroupInfo{
+					Raw:        "",
+					FamilyName: "",
+					GivenName:  "",
+					MiddleName: "",
+					NamePrefix: "",
+					NameSuffix: "",
+				},
+			},
+			RemoveTrailingEmpty: true,
+			IsEmpty:             false,
+		},
+		// Empty with seps
+		{
+			Raw: "=" +
+				"=" +
+				"",
+			Expected: Info{
+				Alphabetic: GroupInfo{
+					Raw:        "",
+					FamilyName: "",
+					GivenName:  "",
+					MiddleName: "",
+					NamePrefix: "",
+					NameSuffix: "",
+				},
+				Ideographic: GroupInfo{
+					Raw:        "",
+					FamilyName: "",
+					GivenName:  "",
+					MiddleName: "",
+					NamePrefix: "",
+					NameSuffix: "",
+				},
+				Phonetic: GroupInfo{
+					Raw:        "",
+					FamilyName: "",
+					GivenName:  "",
+					MiddleName: "",
+					NamePrefix: "",
+					NameSuffix: "",
+				},
+			},
```

Comment:
I tend to err on being super explicit in test cases. While we could elide these fields, including them signals that we KNOW these fields are zero-value, and are in-fact testing them purposefully, no omitting them by accident because they were added after the tests were written.

Reply:
Cool, that's fine!